File: tests/expectations/tag/transparent.c
#include <stdarg.h>
#include <stdbool.h>
#include <stdint.h>
#include <stdlib.h>

struct DummyStruct;

struct EnumWithAssociatedConstantInImpl;

typedef struct DummyStruct TransparentComplexWrappingStructTuple;

typedef uint32_t TransparentPrimitiveWrappingStructTuple;

typedef struct DummyStruct TransparentComplexWrappingStructure;

typedef uint32_t TransparentPrimitiveWrappingStructure;

typedef struct DummyStruct TransparentComplexWrapper_i32;

typedef uint32_t TransparentPrimitiveWrapper_i32;

typedef uint32_t TransparentPrimitiveWithAssociatedConstants;
#define TransparentPrimitiveWithAssociatedConstants_ZERO 0
#define TransparentPrimitiveWithAssociatedConstants_ONE 1

#define EnumWithAssociatedConstantInImpl_TEN 10

void root(TransparentComplexWrappingStructTuple a,
          TransparentPrimitiveWrappingStructTuple b,
          TransparentComplexWrappingStructure c,
          TransparentPrimitiveWrappingStructure d,
          TransparentComplexWrapper_i32 e,
          TransparentPrimitiveWrapper_i32 f,
          TransparentPrimitiveWithAssociatedConstants g,
          struct EnumWithAssociatedConstantInImpl h);

File: tests/expectations/transparent.zig
const std = @import("std");

const DummyStruct = opaque {};

const EnumWithAssociatedConstantInImpl = opaque {};

pub const TransparentComplexWrappingStructTuple = DummyStruct;

pub const TransparentPrimitiveWrappingStructTuple = u32;

pub const TransparentComplexWrappingStructure = DummyStruct;

pub const TransparentPrimitiveWrappingStructure = u32;

pub const TransparentComplexWrapper_i32 = DummyStruct;

pub const TransparentPrimitiveWrapper_i32 = u32;

pub const TransparentPrimitiveWithAssociatedConstants = u32;
pub const TransparentPrimitiveWithAssociatedConstants_ZERO = 0;
pub const TransparentPrimitiveWithAssociatedConstants_ONE = 1;

pub const EnumWithAssociatedConstantInImpl_TEN = 10;

extern fn root(a: TransparentComplexWrappingStructTuple, b: TransparentPrimitiveWrappingStructTuple, c: TransparentComplexWrappingStructure, d: TransparentPrimitiveWrappingStructure, e: TransparentComplexWrapper_i32, f: TransparentPrimitiveWrapper_i32, g: TransparentPrimitiveWithAssociatedConstants, h: EnumWithAssociatedConstantInImpl) anyopaque;
